File: backend/app/models/course.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum


class PaymentType(str, Enum):
    MONTHLY = "monthly"
    PER_LESSON = "per_lesson"
    MIXED = "mixed"


class Course(BaseModel):
    id: str
    title: str
    description: Optional[str] = None
    days: list[str] = []
    time: Optional[str] = None
    price: float = 0.0
    teacher_id: Optional[int] = None
    color: str = "#6C5CE7"
    student_ids: list[str] = []
    location: Optional[str] = None
    location_link: Optional[str] = None
    is_active: bool = True
    created_at: str = ""

    # Tariff fields
    monthly_price: Optional[float] = None
    lesson_price: Optional[float] = None
    lessons_per_week: Optional[int] = None
    payment_type: PaymentType = PaymentType.MONTHLY

File: shared/types/course.ts
export interface Course {
  id: string;
  title: string;
  description?: string;
  days: string[];
  time?: string;
  duration?: number;         // minutes, e.g. 60 or 90
  price: number;
  teacher_id?: number;
  color: string;
  student_ids: string[];
  location?: string;
  location_link?: string;
  is_active: boolean;
  created_at: string;

  // Tariff fields
  monthly_price?: number;
  lesson_price?: number;
  lessons_per_week?: number;
  payment_type?: 'monthly' | 'per_lesson' | 'mixed';
}
